### PR TITLE
fix(obstacle_cruise_planner): improve cruise planner handling to a VRU maneuver overlapping with ego vehicle trajectory

### PR DESCRIPTION
**Pull request by ahmeddesokyebrahim**
## Description
Fixes :
- https://github.com/autowarefoundation/autoware.universe/issues/8311

## Related links
Parent Issue :
- https://github.com/autowarefoundation/autoware.universe/issues/7485

### Before this PR :

https://github.com/autowarefoundation/autoware.universe/assets/36835765/b913905d-eacb-45dc-87d0-1d6effd8d960

https://github.com/autowarefoundation/autoware.universe/assets/36835765/f5367470-0f39-446e-8432-b2780a1098c5

### After this PR :

https://github.com/user-attachments/assets/842ef857-156d-4463-8d0d-0bbdd01b7e7e

https://github.com/user-attachments/assets/47e581e6-676b-4ced-b584-2baa3c8d14f1

### ROS Topic Changes
N.A

### ROS Parameter Changes
| Parameter Name       | Default Value | Update Description                                  | 
| -------------------- | ------------- | --------------------------------------------------- | 
| `obstacle_velocity_threshold_from_cruise_to_stop` | `1.0 m/s`         | Lowering the this threshold below `obstacle_velocity_threshold_from_stop_to_cruise` to prevent chattering between cruise and stop |
| `obstacle_velocity_threshold_from_stop_to_cruise` | `1.5 m/s`         | Lowering the the velocity threshold hysteresis from cruise to obstacle stop so that `obstacle_cruise_planner` is not performing high -ve acceleration once the VRU is overlapping with ego trajectory and allowing the ego vehicle to cruise the front VRU not always triggering obstacle stop for it|

## Effects on system behavior
Enhancing the obstacle_cruise_planner to effectively handle scenarios where vulnerable road users (VRUs) suddenly maneuver and overlap with the ego vehicle's trajectory in dense urban operational design domains (ODDs) will improve deceleration and velocity adjustments, resulting in more human-like driving behavior.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_cruise_planner/obstacle_cruise_planner.param.yaml [hunk 1]
```diff
@@ -89,7 +89,7 @@
 
       # hysteresis for cruise and stop
       obstacle_velocity_threshold_from_cruise_to_stop : 3.0 # stop planning is executed to the obstacle whose velocity is less than this value [m/s]
-      obstacle_velocity_threshold_from_stop_to_cruise : 3.5 # stop planning is executed to the obstacle whose velocity is less than this value [m/s]
+      obstacle_velocity_threshold_from_stop_to_cruise : 2.5 # stop planning is executed to the obstacle whose velocity is less than this value [m/s]
```
Comment (takayuki5168) [inline autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_cruise_planner/obstacle_cruise_planner.param.yaml:92]: `obstacle_velocity_threshold_from_stop_to_cruise` has to be larger than `obstacle_velocity_threshold_from_cruise_to_stop`.
I guess when the ego's velocity is 2.8, the mode will chatter like cruise -> stop -> cruise -> stop -> ...

Comment (ahmeddesokyebrahim) [reply autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_cruise_planner/obstacle_cruise_planner.param.yaml]: Thanks @takayuki5168 -san for your review comment.
I have discussed this comment with @brkay54 as he had previous knowledge in this point and we come to the following updates in the parameters .
- We see that having `obstacle_velocity_threshold_from_stop_to_cruise` with 1.5 m/s (5.4 k/h) would be better handling this scenario allowing the ego vehicle to cruise the VRU not always triggering obstacle stop imitating natural human driving in such dense urban cases
- As well having `obstacle_velocity_threshold_from_cruise_to_stop` less than `obstacle_velocity_threshold_from_stop_to_cruise` will prevent chattering.

I have provided new demo videos under **After this PR**. Please have a look and let us know your thoughts.

Comment (takayuki5168) [reply autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_cruise_planner/obstacle_cruise_planner.param.yaml]: @ahmeddesokyebrahim 
Thank you for the explanation.

In the use case where the ego tries to stop for the front stopped object, decreasing the parameter makes the ego late to switch from CRUISE to STOP, and as a result, the ego is late to decelerate which we feel is very dangerous. So I don't agree with decreasing this parameter.
Does this explanation make sense to you?

Comment (ahmeddesokyebrahim) [reply autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_cruise_planner/obstacle_cruise_planner.param.yaml]: Thanks @takayuki5168 -san for your reply.

> decreasing the parameter makes the ego late to switch from CRUISE to STOP

Actually it is intended by this change to let the ego vehicle cruise the front object not to trigger obstacle stop for it, trying to have a natural human driving behavior in such dense urban areas.

Do you have in mind or with prior experiments any minimum threshold that we can have for `obstacle_velocity_threshold_from_cruise_to_stop` & `obstacle_velocity_threshold_from_stop_to_cruise` ?

**Comment by ahmeddesokyebrahim** [comment]
Closing this PR as the intended issue and scenario/s will be covered by this [PR](https://github.com/autowarefoundation/autoware_launch/pull/1166)

**Pull request closed without merging**